app/controllers/admin/users_controller.rb

```diff
@@ -1,20 +1,24 @@
-class Admin::UsersController < ApplicationController
-  before_action :require_admin
+class Admin::UsersController < AdminController
 
   def index
     @users = User.all
   end
 
-  def destroy
-    User.find(params[:id]).destroy
-    flash[:notice] = "User deleted successfully."
+  def update
+    @user = User.find(params[:id])
+    if @user.admin
```

Comment:
Could this stuff go into a `toggle_admin_privileges` method on the `User` model? Then you could pass back the text for the flash message and shorten up the controller method.